docs/Reference/API Reference - ProMotion CollectionScreen.md

```diff
@@ -0,0 +1,192 @@
+
+    @imageview = UIImageView.new
+    @imageview.frame = [[10, 10], [130, 130]]
+    @imageview.contentMode = UIViewContentModeScaleAspectFit
+    self.contentView.addSubview(@imageview)
+  end
+
+  def setup(cell_data, screen)
+    super
+    url = NSURL.URLWithString(cell_data[:image])
+    @imageview.image = UIImage.imageWithData(NSData.dataWithContentsOfURL(url))
+  end
+end
+
+class PhotosScreen < ProMotion::CollectionScreen
+
+  collection_layout UICollectionViewFlowLayout,
+                    direction:                 :vertical,
+                    minimum_line_spacing:      10,
+                    minimum_interitem_spacing: 10,
+                    item_size:                 [150, 150],
+                    section_inset:             [10, 10, 10, 10]
+
+  cell_classes image_cell: PhotoViewCell
+
+  def collection_data
+    images = [
+      'http://url.com/an_image.jpg',
+    ]
+
+    [
+      images.map do |image|
+        {image: image, cell_identifier: :image_cell}
+      end
+    ]
+  end
+
+end
+```
+
+### Methods
+
+#### collection_data
+
+Method that is called to get the collection's cell data and build the table.
+
+It consists of an array of array of cells.
+
+```ruby
+def collection_data
+  [
+    [
+      { title: 'Oregon', action: :visit_state, arguments: { state: @oregon }},
+      { title: "Washington", action: :visit_state, arguments: { state: @washington }}
+    ],
+    [
+      { title: 'Idaho', action: :visit_state, arguments: { state: @idaho }},
+      { title: "Texas", action: :visit_state, arguments: { state: @texas }}
+    ]
+  ]
+end
+
+def visit_state(args={})
+  mp args[:state] # => instance of State
+end
+```
+
+View the [Reference: All available collection_data options](https://github.com/clearsightstudio/ProMotion/wiki/Reference:-All-available-collection_data-options) for an example with all available options.
+
+#### update_table_data
+
+Causes the table data to be refreshed, such as when a remote data source has
+been downloaded and processed.
+
+```ruby
+class MyCollectionScreen < PM::CollectionScreen
+
+  def on_load
+    MyItem.pull_from_server do |items|
+      @collection_data = [
+        items.map do |item|
+          {
+            title: item.name,
+            action: :tapped_item,
+            arguments: { item: item }
+          }
+        end
+      ]
+
+      update_collection_data
+    end
+  end
+
+  def collection_data
+    @collection_data ||= []
+  end
+
+  def tapped_item(item)
+    open ItemDetailScreen.new(item: item)
+  end
+
+end
+```
+
+#### on_created
+
+Called when a cell is created (not dequeued).  
+
+```ruby
+def on_created
+  self.backgroundColor = UIColor.whiteColor
+
+  @imageview = UIImageView.new
+  @imageview.frame = [[10, 10], [130, 130]]
+  @imageview.contentMode = UIViewContentModeScaleAspectFit
+  self.contentView.addSubview(@imageview)
+end
+```
+
+#### on_reused
+
+Called when a cell is dequeued and re-used. 
+
+#### on_setup(data_cell, screen)
```

Comment:
Mhhh... I did not create this file... some kind of bad merge I guess... I'll change the doc now

About tests, I just ran them again, and yes, they pass locally